Sources/NIO/EventLoop.swift

```diff
@@ -791,3 +763,77 @@ public enum EventLoopError: Error {
     /// Shutting down the `EventLoop` failed.
     case shutdownFailed
 }
+
+fileprivate struct TaskQueue {
```

Comment:
this should have some documentation (and presumably a better name?). This is quite a specialised thing which requires the caller to follow a fairly delicate protocol for it to work correctly which should be documented (and maybe somewhat reflected in the name).